---
title: "Mapping and Counting"
author: "Cheryl Logan"
date: "4/4/2018"
output: html_document
---

## Mapping to the Copper de novo transcriptome

Use RSEM and bowtie for mapping on treebeard. Be sure you are in the QC folder and the samples_file_trimmo is located within it

```{bash, eval=FALSE}
perl /opt/trinityrnaseq-Trinity-v2.4.0/util/align_and_estimate_abundance.pl \       # name of program
--transcripts /home/data/rockfish/assemblies/Copper_Trinity_LT.fasta --seqType fq \ # assembly and seq type
--samples_file /home/data/rockfish/EXP10_Scaur_fluct_gill/QC/samples_file_trimmo \  # samples_file location
--est_method RSEM --aln_method bowtie --trinity_mode \                              # counting and mapping method
--output_dir rsem_outdir                                                            # output directory
```

## Build Transcript and Gene Expression Matrices

```{bash, eval=FALSE}
perl /opt/trinityrnaseq-Trinity-v2.4.0/util/abundance_estimates_to_matrix.pl \
--est_method RSEM \
control_rep1/RSEM.genes.results control_rep2/RSEM.genes.results control_rep3/RSEM.genes.results control_rep4/RSEM.genes.results \
moderate_rep1/RSEM.genes.results moderate_rep2/RSEM.genes.results moderate_rep3/RSEM.genes.results moderate_rep4/RSEM.genes.results \
extreme_rep1/RSEM.genes.results extreme_rep2/RSEM.genes.results extreme_rep3/RSEM.genes.results extreme_rep4/RSEM.genes.results \
relaxation_rep1/RSEM.genes.results relaxation_rep2/RSEM.genes.results relaxation_rep3/RSEM.genes.results relaxation_rep4/RSEM.genes.results \
upwelling_rep1/RSEM.genes.results upwelling_rep2/RSEM.genes.results upwelling_rep3/RSEM.genes.results upwelling_rep4/RSEM.genes.results \
