---
title: "Mapping and Counting"
author: "Cheryl Logan"
date: "4/4/2018"
output: html_document
---

## Mapping to the Copper de novo transcriptome

Use RSEM and bowtie for mapping on treebeard. Be sure you are in the QC folder and the samples_file_trimmo is located within it. Note that we will run all these commands on the treebeard server.

```{bash, eval=FALSE}
perl /opt/trinityrnaseq-Trinity-v2.4.0/util/align_and_estimate_abundance.pl \       # name of program
--transcripts /home/data/rockfish/assemblies/Copper_Trinity_LT.fasta --seqType fq \ # assembly and seq type
--samples_file /home/data/rockfish/EXP10_Scaur_fluct_gill/QC/samples_file_trimmo \  # samples_file location
--est_method RSEM --aln_method bowtie --trinity_mode \                              # counting and mapping method
--output_dir rsem_outdir                                                            # output directory
```

## Build Transcript and Gene Expression Matrices

first make a file.listing_target_files.txt using `nano`

```{bash, eval=FALSE}
perl /opt/trinityrnaseq-Trinity-v2.4.0/util/abundance_estimates_to_matrix.pl \
--est_method RSEM \
--name_sample_by_basedir \   # name 'RSEM.genes.results' file by base directory
control_rep1/RSEM.genes.results control_rep2/RSEM.genes.results control_rep3/RSEM.genes.results control_rep4/RSEM.genes.results \
moderate_rep1/RSEM.genes.results moderate_rep2/RSEM.genes.results moderate_rep3/RSEM.genes.results moderate_rep4/RSEM.genes.results \
extreme_rep1/RSEM.genes.results extreme_rep2/RSEM.genes.results extreme_rep3/RSEM.genes.results extreme_rep4/RSEM.genes.results \
relaxation_rep1/RSEM.genes.results relaxation_rep2/RSEM.genes.results relaxation_rep3/RSEM.genes.results relaxation_rep4/RSEM.genes.results \
upwelling_rep1/RSEM.genes.results upwelling_rep2/RSEM.genes.results upwelling_rep3/RSEM.genes.results upwelling_rep4/RSEM.genes.results 
```

## Compare Replicates
```{bash, eval=FALSE}
perl /opt/trinityrnaseq-Trinity-v2.4.0/Analysis/DifferentialExpression/PtR \
--matrix matrix.counts.matrix \
--samples samples_file --CPM --log2 \
--min_rowSums 10 \
--compare_replicates
```

## Compare Replicates Across Samples
```{bash, eval=FALSE}
perl /opt/trinityrnaseq-Trinity-v2.4.0/Analysis/DifferentialExpression/PtR \
--matrix Trinity_trans.counts.matrix \
--min_rowSums 10 \
-s samples_file --log2 --CPM --sample_cor_matrix
```

## Generate PCA Plot
```{bash, eval=FALSE}
perl /opt/trinityrnaseq-Trinity-v2.4.0/Analysis/DifferentialExpression/PtR \
--matrix matrix.counts.matrix \
-s samples_file \
--min_rowSums 10 --log2 \
--CPM --center_rows \
--prin_comp 3 
```

## Run Differential Gene Expression Analysis with edgeR
```{bash, eval=FALSE}
perl /opt/trinityrnaseq-Trinity-v2.4.0/Analysis/DifferentialExpression/run_DE_analysis.pl  \
--matrix matrix.counts.matrix \
--method edgeR \
--samples_file samples_file
```

## Identify DE genes with p-val and fold change cut-off
```{bash, eval=FALSE}
perl /opt/trinityrnaseq-Trinity-v2.4.0/Analysis/DifferentialExpression/analyze_diff_expr.pl \
--matrix /home/data/rockfish/EXP10_Scaur_fluct_gill/QC/matrix.TMM.EXPR.matrix \
-P 0.001 \
-C 2 \
--samples /home/data/rockfish/EXP10_Scaur_fluct_gill/QC/samples_file
```

## Download the output files to github 430 repo 

First, open a new terminal window on home computer and use `scp` to secure copy these files to the bio430 github repository

```{bash, eval=FALSE}
scp loga8761s@treebeard.csumb.edu:/home/data/rockfish/EXP10_Scaur_fluct_gill/QC/edgeR.40768.dir/* /Users/loga8761/github/bio430/edgeR.40768.dir/  # also get samples file
```

## From here, we will work in Rstudio 