---
title: "merging"
author: "Cheryl Logan"
date: "4/12/2018"
output: html_document
---

## Extracting annotations for your gene list of interest

To do this in R or RStudio, we will take advantage of the `tidyverse` package. You only need to install the package once, but you will need to load the library every time you open up R or RStudio on your computer. So, first install it using the `install.packages` command, then load the library using the command below:

```{r setup, echo = FALSE, message = FALSE, warning = FALSE}
library(tidyverse) # after install.packages("tidyverse")
```

## Using R or RStudio to merge two files

We've done all this in bash so far, but we can do the same join in R or RStudio. First, open RStudio and read in your .csv files:

1. Read in 2 files into R and assign them to `annolist` and `genelist`
```{r read files, eval=FALSE, message = FALSE, warning = FALSE}
genelist <- read_csv("/Users/loga8761/Desktop/matrix.counts.matrix.control_vs_relaxation.edgeR.DE_results.P0.01_C1.DE.subset.csv")
annolist <- read_csv("/Users/loga8761/Desktop/anno_diffExpr.P0.01_C1.matrix.log2.centered.csv")
```

2. View header row of each file to make sure the contig names are identical. In our case, the first column in both files should be called `ContigName`.

```{r, eval=FALSE}
head(annolist) 
head(genelist)
```
3. Once you confirm the column header is the same for both data frames, extract your `genelist` annotations from `annolist` the `right_join` function.
```{r merge, eval=FALSE, message = FALSE, warning = FALSE}
merged <- right_join(annolist,genelist, by = "ContigName") 
head(merged)
```

4. View the new `merged` data frame containing only the annotations for your `genelist` of interest. Take a look at the new file dimensions and contents using `dim` and `head` to make sure the join worked!

```{r, eval=FALSE}
head(merged)
dim(merged) # get file dimensions
```

# Filtering on a merged data frame

There are only 546 rows in `merged` but only 463 in `genelist`! What is going on? Use `head` to inspect the list. Do you see any duplicate ContigNames in `merged`? Yes, we do. Let's quantify  how many times each ContigName appears in `merged` using the `count` function
```{r, eval=FALSE}
merged %>% count(ContigName)
```

We now see that several ContigNames appear multiple times. These represent different isoforms of the same Contig in the assembly file used to create the annotation file. To simplify our downstream analysis, let's try to build a data frame with only one annotation per gene in `genelist`. 

Let's start by identifying the isoform with the lowest e-value based on BLAST results against the nr database. We can  using the `filter` function to find the contig with the lowest e-value (best hit). Since some ContigNames have 'no_sig_hit' they won't have an e-value to filter on. So, let's also use `filter` to extract only the longest contig when there is 'no_sig_hit'.

```{r, eval=FALSE}
unique <- merged %>% # create new variable called unique
  group_by(ContigName) %>%
  filter(topnrEvalue == min(topnrEvalue)) %>%
  filter(ContigLength == max(ContigLength)) %>%
ungroup(merged)

# use count to see if there are any duplicates left
counts <- unique %>% count(ContigName)
```

# The `filter` function filters out NAs!

After inspecting counts, we no longer have any duplicate ContigNames! Hooray! BUT...we now have fewer unique ContigNames in `unique` that in our original `genelist`. This is not what we were expecting! Let's use `anti_join` to see what differences there are between `genelist` and `unique` in the `ContigName` column.

```{r, eval=FALSE}
missing <- anti_join(genelist, unique, by="ContigName")
head(missing)
```
It looks like there are 6 ContigNames missing from `unique` that are in `genelist`... Let's make sure they are in `merged`.

```{r, eval=FALSE}
missing_merged <- anti_join(merged, unique, by="ContigName")
head(missing_merged)
```
These 6 ContigNames are in 'merged'. That means we didn't accidentally lose them when we did our original `merge`. BUT when we inspect the 'topnrmatch' column we see lots of NAs. It looks like the `filter` function filtered out anything with a NA. At this point, we could try to re-run the `filter` function in a way to avoid this problem. Or we can just decide to remove all the ContigNames with NAs (and therefore no annotations) since they are not very useful to us anyway! 

# Remove contigs with no annotations

Let's also go ahead and get rid of any contigs with 'No_sig_nr_hit' using one more `filter` function. We'll call this `genelist_annos`

```{r, eval=FALSE}
genelist_annos <- merged %>% # create new variable called unique
  group_by(ContigName) %>%
  filter(topnrEvalue == min(topnrEvalue)) %>%
  filter(ContigLength == max(ContigLength)) %>%
  filter(!grepl("No_sig_nr_hit",topnrMatch)) %>%
ungroup(merged)

nrow(genelist_annos) # number of rows
```
Of the 463 contigs in `genelist`, it looks like we have annotations for 322 of them (almost 70%... not bad!). As you are scanning through your annotations, I suggest using the 'nobadwordsnrmatch' column match which includes the top hit for which there was a gene name that is more understandable. Let save this as a new .csv file to your current working directory

```{r, eval=FALSE}
write_csv(genelist_annos, "genelist_annos.csv")
```


Once you are ready to learn more about a particular gene, copy the Uniprot ID from the Uniprotmatch column and paste it into http://www.uniprot.org/ search bar.

# Extract unique Uniprot IDs

One more step while we are here. Let's go ahead and pull out a list of UniProt IDs for those contigs that have them.

```{r, eval=FALSE}
genelist_uniprot <- genelist_annos %>% # create new variable called unique
  group_by(ContigName) %>%
  filter(!grepl("No_Uniprotmatch",Uniprotmatch)) %>%
ungroup(genelist_annos)

nrow(genelist_uniprot) # number of rows
```
We see that 260/463 (56%) contigs in `genelist` have UniProt IDs. Not bad. To see how many of Uniprot IDs are unique and to extract a list of them, we can use `count`. 

```{r, eval=FALSE}
genelist_uniprot_unique <- genelist_uniprot %>%
  select("ContigName",Uniprotmatch) %>%
  count(Uniprotmatch) %>%
ungroup(genelist_uniprot)
```

Save this as a new .csv file to your current working directory

```{r, eval=FALSE}
write_csv(genelist_uniprot_unique, "genelist_uniprot_unique.csv")
```

## Using bash to merge files

1. Convert .txt files to .csv files in bash

The first thing we want to do is make sure the files we want to work with are in .csv format. We have two files: a gene list of interest and a really big annotation file for all the contigs in our assembly. If the files are in text (.txt) files, we need to convert to .csv using bash commands below. This will ensure we are using the same file format (and R likes .csv files). If the files are already in .csv format, skip this step and move on.

`sed 's/ \+/,/g' ifile.txt > ofile.csv`

Example:
```{r, eval=FALSE, message = FALSE, warning = FALSE}
sed 's/ \+/,/g' Exp5_CGH_Trinity_annotated.txt > Exp5_CGH_Trinity_annotated.csv
```
 
2. Sort and join two files in bash

Let's say we want to extract the annotations for just our differentially expressed genes (DGElist). First we have to `sort` each .csv file by column 1. Then, we can `join` by column 1 assuming col 1 in  both files contains `ContigNames`:

```{bash, eval=FALSE, message = FALSE, warning = FALSE}
sort -t , -k 1,1 DGElist.csv > DGElist_sorted.csv
sort -t , -k 1,1 big_annotation_file.csv > big_annotation_file_sorted.csv
join -j1 DGElist_sorted.csv big_annotation_file_sorted.csv > annot_DGE.csv
```
 
Now we have a much smaller file (annot_DGE.csv) containing the annotations for our DGE list only!