---
title: "merging"
author: "Cheryl Logan"
date: "4/12/2018"
output: html_document
---

## Extracting annotations for your gene list of interest

To do this in R or RStudio, we will take advantage of the `tidyverse` package. You only need to install the package once, but you will need to load the library every time you open up R or RStudio on your computer. So, first install it using the `install.packages` command, then load the library using the command below:

```{r setup, echo = FALSE, message = FALSE, warning = FALSE}
library(tidyverse) # after install.packages("tidyverse")
```


## Using bash

1. Convert .txt files to .csv files in bash

The first thing we want to do is make sure the files we want to work with are in .csv format. We have two files: a gene list of interest and a really big annotation file for all the contigs in our assembly. If the files are in text (.txt) files, we need to convert to .csv using bash commands below. This will ensure we are using the same file format (and R likes .csv files). If the files are already in .csv format, skip this step and move on.

`sed 's/ \+/,/g' ifile.txt > ofile.csv`

Example:
```{r, eval=FALSE, message = FALSE, warning = FALSE}
sed 's/ \+/,/g' Exp5_CGH_Trinity_annotated.txt > Exp5_CGH_Trinity_annotated.csv
```
 
2. Sort and join two files in bash

Let's say we want to extract the annotations for just our differentially expressed genes (DGElist). First we have to `sort` each .csv file by column 1. Then, we can `join` by column 1 assuming col 1 in  both files contains `ContigNames`:

```{bash, eval=FALSE, message = FALSE, warning = FALSE}
sort -t , -k 1,1 DGElist.csv > DGElist_sorted.csv
sort -t , -k 1,1 big_annotation_file.csv > big_annotation_file_sorted.csv
join -j1 DGElist_sorted.csv big_annotation_file_sorted.csv > annot_DGE.csv
```
 
Now we have a much smaller file (annot_DGE.csv) containing the annotations for our DGE list only!

## Using R or RStudio

We've done all this in bash so far, but we can do the same join in R or RStudio. First, open RStudio and read in your .csv files:

1. Read in 2 files into R
```{r read files, eval=FALSE, message = FALSE, warning = FALSE}
genelist <- read_csv("/Users/loga8761/Desktop/DGE_list.csv")
annolist <- read_csv("/Users/loga8761/Desktop/anno_diffExpr.P0.01_C1.matrix.log2.centered.csv")
```

2. View the files to make sure the header for the contig names are identical. 

```{r, eval=FALSE}
head(annolist)
head(genelist)
```

3. Extract annotations from another gene list in RStudio using `join`
```{r merge, eval=FALSE, message = FALSE, warning = FALSE}
merged <- left_join(genelist, annolist, by = "ContigName") 
```

4. View the new merged file containing only the annotations for your gene list of interest. Take a look at the new file dimensions and contents to make sure the join worked

```{r, eval=FALSE}
head(merged)
dim(merged)
```

There are only 23 genes in our gene list but 55 in our merged file. What is going on? Use head to inspect the list. Do you see any duplicate ContigNames in the merged list? We want to find the top hit with the lowest eval for each contig. We will do this iteratively on the nr evalue, then the uniprot evalue, then the longest contig. After this, we should have a table that has the same number of rows as our gene list of interest!

```{r, eval=FALSE}
unique <- merged %>%
  group_by(ContigName) %>%
  filter(topnrEvalue == min(topnrEvalue)) %>%
  filter(evalue == min(evalue)) %>%
  filter(ContigLength == max(ContigLength)) %>%
ungroup(merged)

head(unique)
dim(unique)
```


Now, we've essentally done the same thing in R as we did in bash, but in R we don't have to sort before we join. 