---
title: "QC using Trimmomatic"
author: "Cheryl Logan (clogan@csumb.edu)"
date: "4/3/2018"
output: html_document
---

## Run Trimmomatic on a single fastq file 

We will use Trimmomatic parameters based on MacManes et al. 2015:

MacManes, M. D. (2014). On the optimal trimming of high-throughput mRNA sequence data. Frontiers in genetics, 5, 13.

In the following example, we run Trimmomatic on 'example using 'CLMP001A_S52_L005_R1_001.fastq'

```{bash, eval=FALSE}
java -jar /opt/Trimmomatic-0.35/trimmomatic-0.35.jar SE \               # call program using SE option
-phred33 \                                                              # Illumina quality score
CLMP001A_S52_L005_R1_001.fastq CLMP001A_S52_L005_R1_001_trimmo.fastq \  # input/output filenames
ILLUMINACLIP:/opt/Trimmomatic-0.35/adapters/TruSeq3-SE.fa:2:30:10 \     # adapter removal
SLIDINGWINDOW:4:5 LEADING:5 TRAILING:5 MINLEN:25                        # poor quality base removal
```

# This will perform the following steps in this order:
1. Remove Illumina adapters provided in TruSeq3-SE.fa (provided)
note: Trimmomatic will look for seed matches (16 bases) allowing maximally 2 mismatches. 
These seeds will be extended and clipped if in the case of paired end reads a score of 
30 is reached (about 50 bases), or in the case of single ended reads a score of 10 (about 17 bases)
2. Remove leading low quality or N bases (below quality 5)
3. Remove trailing low quality or N bases (below quality 5)
4. Scan the read with a 4 base wide sliding window, cutting when the average quality per base drops below 5
5. Drop reads which are less than 25 bases long after these steps

Trimmomatic manual here:
 http://www.usadellab.org/cms/uploads/supplementary/Trimmomatic/TrimmomaticManual_V0.32.pdf

## Run Trimmomatic on all fastq files in a directory 
(on treebeard server)

We will will use a for loop in bash

```{bash, eval=FALSE}
for i in *.fastq;
do
	echo working with "$i"
	newfile="$(basename $i .fastq)"  # for all files ending in .fastq within a directory
    java -jar /opt/Trimmomatic-0.35/trimmomatic-0.35.jar SE \
    -phred33 ${newfile}.fastq ${newfile}.trimmo.fastq \ # '{newfile}' refers to each file being evlauated
    ILLUMINACLIP:/opt/Trimmomatic-0.35/adapters/TruSeq3-SE.fa:2:30:10 \
    SLIDINGWINDOW:4:5 LEADING:5 TRAILING:5 MINLEN:25 
done;
```


## Please cite Trimmomatic as follows:

Bolger, A. M., Lohse, M., & Usadel, B. (2014). Trimmomatic: a flexible trimmer for Illumina sequence data. Bioinformatics, 30(15), 2114-2120.

